src/test/FuzzerImpl.cpp

```diff
@@ -532,27 +657,39 @@ isBadOverlayFuzzerInput(StellarMessage const& m)
     return m.type() == AUTH || m.type() == ERROR_MSG || m.type() == HELLO;
 }
 
-// Empties "ops" as operations are applied.  If `throwIfTxFails` is `false`,
+// Empties "ops" as operations are applied.  If `isSetupOperation` is `false`,
 // ignores failures and might apply further transactions after failures.
+// If `isSetupOperation` is `true`, throws exceptions on transaction failures,
+// and creates new stored ledger key IDs for any new `LedgerEntry`s created by
+// the applied operations.  (When operations are performed by fuzz testing
+// rather than by setup, the new IDs are created when the operations are
+// unpacked.)
 static void
 attemptToApplyOps(LedgerTxn& ltx, PublicKey const& sourceAccount,
                   xdr::xvector<Operation>::const_iterator begin,
                   xdr::xvector<Operation>::const_iterator end, Application& app,
-                  bool const throwIfTxFails = true)
+                  bool const isSetupOperation = true)
 {
     for (auto beginOpsInThisTx = begin; beginOpsInThisTx != end;)
     {
         auto endOpsInThisTx =
             std::distance(beginOpsInThisTx, end) <= MAX_OPS_PER_TX
                 ? end
                 : begin + MAX_OPS_PER_TX;
+
+        if (isSetupOperation)
+        {
+            FuzzUtils::createLedgerKeyIDsForOps(beginOpsInThisTx,
```

Comment:
I can't help but feel like this approach is inverted. You know exactly what you're creating during set-up, so why try to retroactively scrape that data out of the operations?

Reply:
Jon and I talked a bit about this and I've decided to leave it at this stage because, in particular, we get the `ClaimableBalanceID` from the _result_ of a `CREATE_CLAIMABLE_BALANCE` op.  It's a hash of a blob that includes the index of the operation within the transaction, which would be a pain for each individual client of `attemptToApplyOps()` to keep track of.